### DISCUSSION TITLE
How to use Linux environment variables in plugins?

### DISCUSSION THREAD

**Discussion by 1798972** [Kong/kong#10740] [Help] [answered]
**kong version: 2.2.1**
I have now edited the user's environment variables:
```bash
vim ~/.bash_profile
# edit
export MACHINE_ID='10'
source ~/.bash_profile
```
The output is normal after refreshing
```bash
echo $MACHINE_ID
10
```
I try to do this in the plugin
```lua
--(a and {b} or {c})[1]
local mc_id = os.getenv(env_name_machine_id)
kong.log(" To generate the pipeline, read the environment variable name: ".. env_name_machine_id ..  "| variable value:".. tostring(mc_id))
```
But the value you get is always `nil`.

---

I have another version `2.7.2`, which is fine.
but now I have to use the environment variables in version `2.2.1`
thanks...

**Comment by locao** [comment]
For security reasons, Nginx does not allow access to environment variables, you can read more in [lua-nginx-module docs](https://github.com/openresty/lua-nginx-module#system-environment-variable-support) and [Nginx docs](https://nginx.org/en/docs/ngx_core_module.html#env). 

To enable access to the required environment variable you can add it to [kong/templates/nginx.lua](https://github.com/Kong/kong/blob/2.2.1/kong/templates/nginx.lua) and restart Kong.

**Reply by 1798972** [reply]
Thank you very much for your suggestion. 
Well, I read that this is a lua file, but I still don't quite understand it. Do you have any specific examples...

**Reply by Tieske** [reply, marked as answer]
During startup, when the plugin code gets loaded, the environment variables will still be available. After the engine started, they will no longer be accessible.

so in your `handler.lua`, at the very top, do;
```lua
local MY_VALUE = os.getenv("MY_ENV_VARIABLE_NAME")
```
This will get evaluated during initialization.

In all the code below it, you can now access the module-level variable `MY_VALUE` to access it.
